flink-python/pyflink/fn_execution/coders.py

```diff
@@ -226,13 +226,24 @@ class ArrayCoder(CollectionCoder):
     """
 
     def __init__(self, elem_coder):
-        self._elem_coder = elem_coder
         super(ArrayCoder, self).__init__(elem_coder)
 
     def get_impl(self):
         return coder_impl.ArrayCoderImpl(self._elem_coder.get_impl())
 
 
+class PrimitiveArrayCoder(CollectionCoder):
```

Comment:
Now that we have introduced `PrimitiveArrayCoder`, can we rename `ArrayCoder` to `BasicArrayCoder`